packages/components/src/components/input-adapter-leanup/controller.ts

```diff
@@ -4,58 +4,131 @@ import { validateTouched } from '../../types/props/touched';
 		}
 	}
 
+	protected setAttribute(qualifiedName: string, element?: HTMLElement, value?: string | number | boolean) {
+		if (EXPERIMENTAL_MODE) {
+			try {
+				value = typeof value === 'object' && value !== null ? JSON.stringify(value) : value;
+				if (typeof value === 'boolean' || typeof value === 'number' || typeof value === 'string') {
+					element?.setAttribute(qualifiedName, `${value as string}`);
+				} else {
+					throw new Error(`Invalid value type: ${typeof value}`);
+				}
+			} catch (e) {
+				element?.removeAttribute(qualifiedName);
+			}
+		}
+	}
+
 	/**
 	 * We try to support native form-associated custom elements.
 	 *
 	 * @see https://github.com/public-ui/kolibri/discussions/2821
 	 */
-	protected readonly syncFormAssociatedName = () => {
-		syncElementAttribute('id', this.formAssociated, this.component.state._id as string);
-		syncElementAttribute('name', this.formAssociated, this.component.state._name as string);
-		syncElementAttribute('value', this.formAssociated, this.component.state._value as string);
-	};
-
 	public readonly setFormAssociatedValue = (value: string | null = null) => {
-		syncElementAttribute('value', this.formAssociated, value as string);
-		syncElementAttribute('value', this.syncToOwnInput, value as string);
+		const name = this.formAssociated?.getAttribute('name');
+		if (name === null || name === '') {
+			devHint(` The form field (${this.name}) must have a name attribute to be form-associated. Please define the _name attribute.`);
+		}
+
+		try {
+			/**
+			 * We need to stringify the value, for the setAttribute method.
+			 * @see https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute
+			 * @see https://developer.mozilla.org/en-US/docs/Web/API/Element/attributes
+			 * @see https://developer.mozilla.org/en-US/docs/Web/API/Element/attributes#value
+			 *
+			 * TODO: It is possible that the value are a cyclic object value. So we need a custom
+			 *       JSON.stringify method from outside to convert it to string.
+			 */
+			const val = typeof value === 'object' && value !== null ? JSON.stringify(value) : value;
+			if (typeof val === 'boolean' || typeof val === 'number' || typeof val === 'string') {
+				this.formAssociated?.setAttribute('value', val);
+				this.syncToOwnInput?.setAttribute('value', val);
+			} else {
+				throw new Error(`Invalid value type: ${typeof val}`);
+			}
+			switch (this.name) {
+				case 'select':
+					if (this.formAssociated) {
+						(this.formAssociated as HTMLSelectElement).querySelectorAll('option').forEach((el) => {
+							(this.formAssociated as HTMLSelectElement).removeChild(el);
+						});
+						if (Array.isArray(value) && value.length > 0) {
+							value.forEach((val) => {
+								const option = document.createElement('option');
+								option.setAttribute('value', val as string);
+								option.setAttribute('selected', '');
+								(this.formAssociated as HTMLSelectElement).appendChild(option);
```

Comment:
Wenn das zu oft aufgerufen wird, könnte es in die Performance gehen. Ist es vielleicht besser, vorher zu prüfen, ob das Array verändert wurde und dann erst den DOM manipulieren?

Reply:
wird nur bei value change gemacht